analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -155,9 +155,16 @@ def func_map_list_src_to_ast(func_src_list, ctu_on_demand):
 
     func_ast_list = []
     for fn_src_txt in func_src_list:
-        dpos = fn_src_txt.find(" ")
-        mangled_name = fn_src_txt[0:dpos]
-        path = fn_src_txt[dpos + 1:]
+        if fn_src_txt[0].isdigit():
+            length_str, _ = fn_src_txt.split(':', 1)
+            length = int(length_str)
+            sep_pos = len(length_str) + 1 + length
+            mangled_name = fn_src_txt[0: sep_pos]
```

Comment:
Are you sure that the mangled name starts at the beginning of the line? Is the length part of the mangled name?

Reply:
Well, it seems misleading that we handle the length as part of the mangled_name. However, it doesn't really matter because at the end we just concatenate the two and append it as a string:
```
func_ast_list.append(mangled_name + " " + mapped_path)
```

Reply:
But if you're processing the data into a space-separated list, can the consumer still handle it...? How would or why wouldn't the same issue resurface in a later step of the pipeline? My guess is that this list is now only for CodeChecker to handle internally. Could we pass a list of two-tuples?